addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2033,6 +2039,7 @@ const ListUserValueWidget = UserValueWidget.extend({
      * @private
      */
     _notifyCurrentState() {
+        this._previousValue = this._value;
```

Comment:
I don't like this fix :grimacing: The way the widgets are supposed to work is that their value is supposed to be given to them. From there the user can interact with the widget, which notifies a new value... but then receive the new value it should have (probably the value it notified but maybe not, the option may decide to decide it cannot do anything with it and reset it or whatever). The widget that way always reflect what the DOM shows. So here a widget remembering what it notifies is not good.

> It is a temporary fix for V15 as the Many2manyUserValueWidget may be refactored later.

Why not fixing properly now? :thinking: From the commit message and a quick look at the code, I am not sure why there are rpc being done each time... seems the RPC are done in the `willStart` which should be called only once? :thinking: Also this fix seems to be genereic to m2m widgets and not only the visibility options one... but I don't see many rpc being done with other option using m2m widgets. So unless I am missing something, the fix should be specific to the option and not to the widget?

Reply:
From what I understood, updating a snippet option will update the UI of every options through `_onSnippetOptionUpdate -> editor.updateOptionsUI() -> opt.updateUI({noVisibility: true})`
Then each option will set the value of its widgets `widget.setValue(normalizedValue, methodName)` in its updateUI method.
When a ListUserValueWidget is initialized with a Many2oneUserValueWidget as its createWidget, which is the case for the M2M, the ListUserValueWidget.setValue will trigger an input event on the M2O.inputEl, which will lead to a Many2oneUserValueWidget._search and fire the RPCs.
I am sorry it's that verbose and unclear... but that led me to think that the fix should be done on the M2M/M2O/ListWidgets, as updating any option of the blog cover will trigger two rpc for the tags here https://9975255-master-all.runbot46.odoo.com/blog. What do you think ? I will try to find a fix that is not going against the widget lifecycle, maybe there is some refactoring to do in the ListUserValueWidget.setValue

Reply:
> I am sorry it's that verbose and unclear

Your explanation seems clear to me :wink: 

> as updating any option of the blog cover [...]

Ok then if there are other cases, you are probably right that this is the widget that needs to be fixed. I probably missed something when testing.

> maybe there is some refactoring to do in the ListUserValueWidget.setValue
> the ListUserValueWidget.setValue will trigger an input event on the M2O.inputEl, which will lead to a Many2oneUserValueWidget._search and fire the RPCs

Yes, it's indeed strange that a setValue triggers an input event, there is probably something to do around that code :+1: